### ISSUE TITLE
TypeError: options.locEnd is not a function

### ISSUE THREAD

**Issue by Jackson-soft** [prettier/plugin-lua#20]
rtp.lua[error] rtp.lua: TypeError: locEnd is not a function
[error]     at isNextLineEmpty (/usr/lib/node_modules/prettier/index.js:3572:42)
[error]     at /usr/lib/node_modules/@prettier/plugin-lua/src/printer.js:410:7
[error]     at /usr/lib/node_modules/prettier/index.js:14487:23
[error]     at FastPath.each (/usr/lib/node_modules/prettier/index.js:14474:7)
[error]     at FastPath.map (/usr/lib/node_modules/prettier/index.js:14486:10)
[error]     at printBody (/usr/lib/node_modules/@prettier/plugin-lua/src/printer.js:384:8)
[error]     at printNoParens (/usr/lib/node_modules/@prettier/plugin-lua/src/printer.js:28:14)
[error]     at Object.genericPrint [as print] (/usr/lib/node_modules/@prettier/plugin-lua/src/printer.js:763:19)
[error]     at callPluginPrintFunction (/usr/lib/node_modules/prettier/index.js:14718:18)
[error]     at /usr/lib/node_modules/prettier/index.js:14649:47

**Comment by suchipi** [comment]
Could you provide some more information?

**Comment by Jackson-soft** [comment]
❯ prettier -v
2.2.1

**Comment by rubixibuc** [comment]
I am getting the same error when formatting any file.

1. Installed lua prettier plugin
2. Ran prettier [some file].lua with just the line `local t = {}`
3. Got this error

` TypeError: locEnd is not a function
[error]     at isNextLineEmpty (/Users/jasonrubenstein/Dropbox/__new__/dev/webcomponents/livechalk-micro/node_modules/prettier/index.js:3572:42)
[error]     at /Users/jasonrubenstein/Dropbox/__new__/dev/webcomponents/livechalk-micro/node_modules/@prettier/plugin-lua/src/printer.js:410:7
[error]     at /Users/jasonrubenstein/Dropbox/__new__/dev/webcomponents/livechalk-micro/node_modules/prettier/index.js:14487:23
[error]     at FastPath.each (/Users/jasonrubenstein/Dropbox/__new__/dev/webcomponents/livechalk-micro/node_modules/prettier/index.js:14474:7)
[error]     at FastPath.map (/Users/jasonrubenstein/Dropbox/__new__/dev/webcomponents/livechalk-micro/node_modules/prettier/index.js:14486:10)
[error]     at printBody (/Users/jasonrubenstein/Dropbox/__new__/dev/webcomponents/livechalk-micro/node_modules/@prettier/plugin-lua/src/printer.js:384:8)
[error]     at printNoParens (/Users/jasonrubenstein/Dropbox/__new__/dev/webcomponents/livechalk-micro/node_modules/@prettier/plugin-lua/src/printer.js:28:14)
[error]     at Object.genericPrint [as print] (/Users/jasonrubenstein/Dropbox/__new__/dev/webcomponents/livechalk-micro/node_modules/@prettier/plugin-lua/src/printer.js:763:19)
[error]     at callPluginPrintFunction (/Users/jasonrubenstein/Dropbox/__new__/dev/webcomponents/livechalk-micro/node_modules/prettier/index.js:14718:18)
[error]     at /Users/jasonrubenstein/Dropbox/__new__/dev/webcomponents/livechalk-micro/node_modules/prettier/index.js:14649:47
`

node -v
v14.15.1

**Comment by nathanloyer** [comment]
I had this issue as well when first trying this plugin out on a repo. Guess I'll use something else

**Comment by liuxian1990** [comment]
1.  prettier@2.2.1    3571 line  
```js
function isNextLineEmpty(text, node, locEnd) {
  return isNextLineEmptyAfterIndex(text, locEnd(node));
}
```
2. plugin-lua  410 line 
```js
isNextLineEmpty(text, statement, options)
```
A temporary solution

```js
    function locEnd(node) {
      return node.range[1];
    }

    if (
      // isNextLineEmpty(text, statement, options) &&
      isNextLineEmpty(text, statement, locEnd) &&
      !isLastStatement(statementPath)
    ) {
      parts.push(hardline);
    }

```

**Comment by liuxian1990** [comment]
1.  prettier@2.2.1    3571 line  
```js
function isNextLineEmpty(text, node, locEnd) {
  return isNextLineEmptyAfterIndex(text, locEnd(node));
}
```
2. plugin-lua  410 line 
```js
isNextLineEmpty(text, statement, options)
```
A temporary solution

```js
    function locEnd(node) {
      return node.range[1];
    }

    if (
      // isNextLineEmpty(text, statement, options) &&
      isNextLineEmpty(text, statement, locEnd) &&
      !isLastStatement(statementPath)
    ) {
      parts.push(hardline);
    }

```

**Comment by creepinson** [comment]
Same issue - only other lua formatter I could find is luafmt but it doesn't have editorconfig support or a config file :(

**Comment by Stanzilla** [comment]
https://github.com/JohnnyMorganz/StyLua try this, it does not have a lot of options but maybe it's good enough for you

**Comment by TimSBSquare** [comment]
This appears to be fixed on master but not yet released

**Comment by Jackson-soft** [comment]
Is there a release time? Currently this problem affects the use

**Comment by TimSBSquare** [comment]
Sorry @Jackson-soft I'm not a maintainer

**Comment by tims-bsquare** [comment]
@CleoMenezesJr it currently works on master. You can fork it and push it to npm. I don't think anyone is actively maintaining a fork afaik

**Comment by shmup** [comment]
> https://github.com/JohnnyMorganz/StyLua try this, it does not have a lot of options but maybe it's good enough for you

I want to use this, and I want to configure it in vim via `formatprg`, but I have this ticket up looking for clarification: https://github.com/JohnnyMorganz/StyLua/issues/516

Any clue if I can prevent it from writing to disk, and just to stdout?

**Comment by shmup** [comment]
> @CleoMenezesJr it currently works on master. You can fork it and push it to npm. I don't think anyone is actively maintaining a fork afaik

If you're just trying to make this available on your machine, here's what I did:

`npm install -g git+https://github.com/prettier/plugin-lua.git`

**Comment by Mayron** [comment]
Same error

**Comment by atburghardt** [comment]
> > Well, this issue has been open for a long time. Did anyone find any solution?
> 
> > @CleoMenezesJr it currently works on master. You can fork it and push it to npm. I don't think anyone is actively maintaining a fork afaik
> 
> If you're just trying to make this available on your machine, here's what I did:
> 
> `npm install -g git+https://github.com/prettier/plugin-lua.git`

this works localy. I like to use the working lua-plugin in github actions. What steps are needed for this to be deployed?

**Comment by newit-trungnt** [comment]
> > Well, this issue has been open for a long time. Did anyone find any solution?
> 
> > @CleoMenezesJr it currently works on master. You can fork it and push it to npm. I don't think anyone is actively maintaining a fork afaik
> 
> If you're just trying to make this available on your machine, here's what I did:
> 
> `npm install -g git+https://github.com/prettier/plugin-lua.git`
This is works for me.
Thanks @shmup

**Comment by elsongabriel** [comment]
idk how.. but it's worked for me.

in **plugin-lua/src/printer.js**

I changed (line 410):
`isNextLineEmpty(text, statement, options) &&`
to
`isNextLineEmpty(text, statement, locEnd) &&`

I changed (line 621):
`} else if (isNextLineEmpty(options.originalText, arg, options)) {`
to 
`} else if (isNextLineEmpty(options.originalText, arg, locEnd)) {`

created in line 427
`function locEnd(node) {
	return node.range[1];
}`

then in my project, I created a script in package.json
`"prettier:format": "prettier --write \"**/*.lua\""`


but in some files of my project, i had this problem:

> [error] MYPATH\MYFILE.lua: Error: Comment "MY FILE COMMENT" was not printed. Please report this error!
[error]     at Object.ensureAllCommentsPrinted (C:\MYPATH\node_modules\prettier\index.js:8290:17)
[error]     at coreFormat (C:\MYPATH\node_modules\prettier\index.js:8839:16)
[error]     at formatWithCursor2 (C:\MYPATH\node_modules\prettier\index.js:9021:18)
[error]     at Object.formatWithCursor (C:\MYPATH\node_modules\prettier\index.js:38176:12)
[error]     at format (C:\MYPATH\node_modules\prettier\cli.js:15017:24)
[error]     at formatFiles2 (C:\MYPATH\node_modules\prettier\cli.js:15131:22)
[error]     at async main (C:\MYPATH\node_modules\prettier\cli.js:15356:5)
[error]     at async Object.run (C:\MYPATH\node_modules\prettier\cli.js:15303:5)

**Comment by suchipi** [comment]
This has been fixed in `@prettier/plugin-lua` version 0.0.2
